Common/Base/SDL2/BaseManager.cpp

```diff
@@ -0,0 +1,109 @@
+#include "Precompiled.h"
+#include "BaseManager.h"
+
+#include <SDL_image.h>
+
+namespace base
+{
+	bool BaseManager::createRender(int _width, int _height, bool _windowed)
+	{
+		mRenderer = SDL_CreateRenderer(mSdlWindow, 0, SDL_RENDERER_ACCELERATED | SDL_RENDERER_TARGETTEXTURE | SDL_RENDERER_PRESENTVSYNC);
+
+		if (mRenderer == nullptr) {
+			std::cerr << "Failed to initialize SDL_Renderer: " << SDL_GetError();
+			exit(1);
+		}
+
+		// TODO: Why do we need this? We need to understand why SubSkin::doRender() multiplies the size of the quad by 2
+//		SDL_RenderSetScale(mRenderer, 0.5, 0.5);
+
+		if (IMG_Init(~0) == 0)
+		{
+			std::cerr << "Failed to initialize SDL_image: " << IMG_GetError();
+			exit(1);
+		}
+		return true;
+	}
+
+	void BaseManager::destroyRender()
+	{
+		SDL_DestroyRenderer(mRenderer);
+		IMG_Quit();
+	}
+
+	void BaseManager::createGuiPlatform()
+	{
+		mPlatform = new MyGUI::SDL2Platform();
+		setupResources();
+		mPlatform->initialise(this, mRenderer);
+	}
+
+	void BaseManager::destroyGuiPlatform()
+	{
+		if (mPlatform)
+		{
+			mPlatform->shutdown();
+			delete mPlatform;
+			mPlatform = nullptr;
+		}
+	}
+
+	void BaseManager::drawOneFrame()
+	{
+		SDL_SetRenderDrawColor(mRenderer, 10, 20, 30, 0xFF);
+		SDL_RenderClear(mRenderer);
+
+		if (mPlatform)
+			mPlatform->getRenderManagerPtr()->drawOneFrame();
+
+		SDL_RenderPresent(mRenderer);
+	}
+
+	void BaseManager::resizeRender(int _width, int _height)
+	{
+		// TODO: Maybe we need to resize the renderer?
+		printf("Resized\n");
```

Comment:
- [ ] I actually need to verify if SDL2 takes care of this automatically or we actually need to manually do it